### ISSUE TITLE
Icons resolution issue

### ISSUE THREAD

**Issue by Lija321** [costales/folder-color#38]
Im running Ubuntu 22.04.3 LTS with nautils and wayland, when I installed the extention is used the 48x48 icons, I fixed it by moving the 256x256 icons to the 48x48 folder but i reckon there is a better soulution

**Comment by Arnwaltur** [comment]
Same on Ubuntu 23.10 + wayland.
And icons are slightly blurred with screen scaling.

**Comment by costales** [comment]
Hi, probably is about the icon theme you would use. Could you tell me the icon theme and from where you installed it (PPA, Ubuntu repos, etc)? Thanks!

**Comment by Arnwaltur** [comment]
Hi, use Yaru default icon theme from Ubuntu 23.10 fresh install.

**Comment by costales** [comment]
> Hi, use Yaru default icon theme from Ubuntu 23.10 fresh install.

Hi, I suppose you're using the default folder-color from Ubuntu repository then. 

Please, try this steps:

1. Uninstall all `folder-color*` packages (include the '*').
2. Add the PPA: `sudo add-apt-repository ppa:costales/folder-color ; sudo apt update`
3. Install just `folder-color` package.

In my latest checks, Yaru doesn't include folder-color icons.

**Comment by Arnwaltur** [comment]
Thanks for your help.
I followed your steps, but it doesn't change anything, even after a reboot.

After checking in : /usr/share/icons/Yaru 
Folders /places contains folder-color icons.

I tried to modify /usr/share/nautilus-python/extensions/folder-color.py in order to change resolution and scaling from line :
icon = icon_theme.lookup_icon(...)
But it doesn't seem to make any difference, even after nautilus relaunch or reboot.

But it's not that serious, it's purely aesthetic.

**Comment by costales** [comment]
Thanks for the feedback.
Could you run? dpkg -l | grep yaru
Best regards

**Comment by costales** [comment]
Downloading Yaru icons from the [Ubuntu repo](https://packages.ubuntu.com/lunar/yaru-theme-icon) I confirm that they don't include support for Folder Color extension:

```
$ ls
Yaru       Yaru-bark-dark  Yaru-blue-dark  Yaru-magenta       Yaru-olive       Yaru-prussiangreen       Yaru-purple       Yaru-red       Yaru-sage       Yaru-viridian
Yaru-bark  Yaru-blue       Yaru-dark       Yaru-magenta-dark  Yaru-olive-dark  Yaru-prussiangreen-dark  Yaru-purple-dark  Yaru-red-dark  Yaru-sage-dark  Yaru-viridian-dark
$ find . -name "*folder-blue*"
$ 
```

**Comment by Arnwaltur** [comment]
$ dpkg -l | grep yaru
ii  libreoffice-style-yaru                        4:7.6.4-0ubuntu0.23.10.1                all          office productivity suite -- Yaru symbol style
ii  yaru-theme-gnome-shell                        23.10.0-0ubuntu2                        all          Yaru GNOME Shell desktop theme from the Ubuntu Community
ii  yaru-theme-gtk                                23.10.0-0ubuntu2                        all          Yaru GTK theme from the Ubuntu Community
ii  yaru-theme-icon                               23.10.0-0ubuntu2                        all          Yaru icon theme from the Ubuntu Community
ii  yaru-theme-sound                              23.10.0-0ubuntu2                        all          Yaru sound theme from the Ubuntu Community


$ find . -name "*folder-blue*"
./Yaru/16x16/places/folder-blue.png
./Yaru/32x32@2x/places/folder-blue.png
./Yaru/24x24/places/folder-blue.png
./Yaru/256x256/places/folder-blue-public.png
./Yaru/256x256/places/folder-blue-documents.png
./Yaru/256x256/places/folder-blue-music.png
./Yaru/256x256/places/folder-blue-videos.png
./Yaru/256x256/places/folder-blue-downloads.png
./Yaru/256x256/places/folder-blue.png
./Yaru/256x256/places/folder-blue-templates.png
./Yaru/256x256/places/folder-blue-pictures.png
./Yaru/48x48@2x/places/folder-blue-public.png
./Yaru/48x48@2x/places/folder-blue-documents.png
./Yaru/48x48@2x/places/folder-blue-music.png
./Yaru/48x48@2x/places/folder-blue-videos.png
./Yaru/48x48@2x/places/folder-blue-downloads.png
./Yaru/48x48@2x/places/folder-blue.png
./Yaru/48x48@2x/places/folder-blue-templates.png
./Yaru/48x48@2x/places/folder-blue-pictures.png
./Yaru/32x32/places/folder-blue.png
./Yaru/48x48/places/folder-blue-public.png
./Yaru/48x48/places/folder-blue-documents.png
./Yaru/48x48/places/folder-blue-music.png
./Yaru/48x48/places/folder-blue-videos.png
./Yaru/48x48/places/folder-blue-downloads.png
./Yaru/48x48/places/folder-blue.png
./Yaru/48x48/places/folder-blue-templates.png
./Yaru/48x48/places/folder-blue-pictures.png
./Yaru/256x256@2x/places/folder-blue-public.png
./Yaru/256x256@2x/places/folder-blue-documents.png
./Yaru/256x256@2x/places/folder-blue-music.png
./Yaru/256x256@2x/places/folder-blue-videos.png
./Yaru/256x256@2x/places/folder-blue-downloads.png
./Yaru/256x256@2x/places/folder-blue.png
./Yaru/256x256@2x/places/folder-blue-templates.png
./Yaru/256x256@2x/places/folder-blue-pictures.png
./Yaru/24x24@2x/places/folder-blue.png
./Yaru/16x16@2x/places/folder-blue.png

I just followed the instructions in the README for installation.

**Comment by costales** [comment]
Hi,

I see that folder-color package doesn't exists in Ubuntu 23.10 repos (weird).

I downloaded the source of Yaru icon theme in Ubuntu 23.10 and it hasn't any `*folder-blue*` file:

```
$ find . -name *folder-blue*
$ ls
AUTHORS          COPYING           COPYING.LGPL-3.0  README.md     cinnamon-shell  create-new-variant.md  gnome-shell  gtksourceview  meson.build        metacity  sounds        xfwm4
CONTRIBUTING.md  COPYING.LGPL-2.1  LICENSE_CCBYSA    bootstrap.sh  common          debian                 gtk          icons          meson_options.txt  sessions  ubuntu-unity
$
```

I would like to test your exactly installation. Could you tell me what I need? Ubuntu 23.10 + yaru-theme-icon from repository + folder-color from repository? Thanks.

**Comment by Arnwaltur** [comment]
Hi,

I agree. 
All folder-color are not present as standard, it's the costales/folder-color/setup.py installation script that adds it in /usr/share/icons/Yaru/.../places

**Comment by costales** [comment]
Found the issue... Checking for a solution. Really thank you for your feedback!

**Comment by costales** [comment]
Hi! Please, could you confirm it's working with the release 0.4.1 from the PPA? (Uninstall `folder-color*` first) Thank you!

**Comment by Arnwaltur** [comment]
Things are looking up. It's sharper. For my use, it's ok.
Thank you very much.

However, there's still a problem when you change the size of the icons in nautilus. I've added a video capture to illustrate.

[folder-color-0.4.1.webm](https://github.com/costales/folder-color/assets/63635642/d40ab0ac-51ca-41e2-a168-58db2c4c8c8f)

**Comment by costales** [comment]
Thanks for your feedback!

I just tried your test and I can't reproduce it. You were in Ubuntu 23.10 right? This is my test:

[Screencast from 2024-01-22 19-46-17.webm](https://github.com/costales/folder-color/assets/5886786/496eedff-f812-4c18-91fe-fa3f34e52f35)

Someone could confirm this please?

**Comment by DiogoConstantino** [comment]
I've just tried it, also with a fresh installation of Ubuntu 23.10 (a VM), and Wayland, and I've successfully reproduced the issue.

![imagem](https://github.com/costales/folder-color/assets/9408012/90932e6b-dfb4-40e9-8186-828f18292b1d)

**Comment by Arnwaltur** [comment]
Ubuntu 23.10 on a 4K monitor with 200% scaling.

**Comment by QuentiumYT** [comment]
I actually found my folders blurred as well, I had to reapply the folder's color again and everything is fine. Thanks for the fix.
![image](https://github.com/costales/folder-color/assets/35229942/368fbf8d-6113-42ec-8ee4-a96032b98af7)


Just a question, since last version 0.3.2, the folder was all green and not only the top, may I know why it changed? I like both tho

**Comment by QuentiumYT** [comment]
Oh alright didn't know this behaviour, I thought it was doing it globally and then look for the best size :)
I see, this is looking more uniformized with the current folder icon. The previous version for me was great as it was outstanding, what about keeping both @costales? Thanks for this great project tho!

**Comment by costales** [comment]
If you prefer the previous icons, this theme would fit it :)
https://github.com/pakaoraki/yaru-colors-folder-color
--
Marcos Costales.

**Comment by vescovaccio** [comment]
I tried to follow this thread, and I still have fuzzy icons, from minimum to maximum zoom in Files, and the folder becoming giant and larger than all the others when I check the properties. This is on Ubuntu 23.10 + Wayland + Gnome 47. I don't know whether there is a difference here between Wayland and X, but I didn't have this before when I was on X. I tried all of the various suggestions listed above, including copying the 256x256 icons to the 48x48 and 32x32 folders, and trying to install Pakaoraki's alternative icons - which simply gave me a dpkg error and wouldn't install.

After all the testing, and leaving folder-color installed, this is the result of "$ dpkg -l | grep yaru":

ii  libreoffice-style-yaru                        4:24.8.2-0ubuntu1                        all          office productivity suite -- Yaru symbol style
ii  yaru-theme-gnome-shell                        24.10.4-0ubuntu1                         all          Yaru GNOME Shell desktop theme from the Ubuntu Community
ii  yaru-theme-gtk                                24.10.4-0ubuntu1                         all          Yaru GTK theme from the Ubuntu Community
ii  yaru-theme-icon                               24.10.4-0ubuntu1                         all          Yaru icon theme from the Ubuntu Community
ii  yaru-theme-sound                              24.10.4-0ubuntu1                         all          Yaru sound theme from the Ubuntu Community
